dialoguekit/nlg/template_from_training_data.py

```diff
@@ -27,7 +27,7 @@ def _replace_slot_with_placeholder(
         annotated_utterance._text = annotated_utterance.text.replace(
```

Comment:
There should be a setter for `text` in AnnotatedUtterance, instead of changing member variables directly. (If it's converted to a dataclass than that should take care of this.)

Reply:
Will be done in a follow-up issue #188